### PR TITLE
Fixes for cppfront

### PR DESCRIPTION
**Pull request by mattgodbolt**
This also does a TODO about output monaco assembly modes.

See #4074

### CODE REVIEW

**Comment by mattgodbolt** [comment]
CC @hsutter - will merge in the morning

File: lib/languages.ts [hunk 1]
```diff
@@ -192,16 +216,18 @@ const definitions: Record<LanguageKey, LanguageDefinition> = {
...
         previewFilter: null,
+        monacoDisassembly: null,
     },
-    cppx_cppfront: {
-        name: 'Cppx-cppfront',
-        monaco: 'cppx-cppfront',
+    cpp2_cppfront: {
+        name: 'Cpp2-cppfront',
+        monaco: 'cpp2-cppfront',
         extensions: ['.cpp2'],
```
Comment (partouf) [inline lib/languages.ts:224]: shouldn't this be .cpp, or does that give issues?

File: static/panes/compiler.js [hunk 2]
```diff
@@ -128,21 +128,8 @@ function Compiler(hub, container, state) {
...
-            case 'ruby':
-                monacoDisassembly = 'asmruby';
-                break;
-            case 'mlir':
-                monacoDisassembly = 'mlir';
-                break;
-        }
-    }
+    var monacoDisassembly =
+        (languages[this.currentLangId] ? languages[this.currentLangId].monacoDisassembly : null) || 'asm';
```
Comment (partouf) [inline static/panes/compiler.js:132]: todo: this should actually be a runtime thing and depend on the compilationresult. But this is the next best thing, so thumbs up from me

File: lib/languages.ts (hunk 1 above)
Comment (mattgodbolt) [reply lib/languages.ts:224]: Herb wanted it called "Cpp2" vs "Cpp" vs "Cppx".

File: static/panes/compiler.js (hunk 2 above)
Comment (mattgodbolt) [reply static/panes/compiler.js:132]: Right, thanks, yeah :)

**Pull request merged**